src/components/v2/V2Project/ProjectActivity/index.tsx

```diff
@@ -0,0 +1,368 @@
+        keys: ['id', 'txHash', 'timestamp', 'caller'],
+      },
+      {
+        entity: 'distributePayoutsEvent',
+        keys: [
+          'id',
+          'timestamp',
+          'txHash',
+          'caller',
+          'beneficiary',
+          'distributedAmount',
+          'memo',
+        ],
+      },
+      {
+        entity: 'distributeReservedTokensEvent',
+        keys: [
+          'id',
+          'timestamp',
+          'txHash',
+          'caller',
+          'beneficiary',
+          'beneficiaryTokenCount',
+          'tokenCount',
+        ],
+      },
+    ],
+    orderDirection: 'desc',
+    orderBy: 'timestamp',
+    where,
+  })
+
+  const list = useMemo(
+    () =>
+      projectEvents?.pages.map(group =>
+        group.map((e: ProjectEvent) => {
+          let elem: JSX.Element | undefined = undefined
+
+          if (e.payEvent) {
+            elem = <PayEventElem event={e.payEvent as PayEvent} />
+          }
+          if (e.redeemEvent) {
+            elem = <RedeemEventElem event={e.redeemEvent as RedeemEvent} />
+          }
+          if (e.projectCreateEvent) {
+            elem = (
+              <ProjectCreateEventElem
+                event={e.projectCreateEvent as ProjectCreateEvent}
+              />
+            )
+          }
+          if (e.deployedERC20Event) {
+            elem = (
+              <DeployedERC20EventElem
+                event={e.deployedERC20Event as DeployedERC20Event}
+              />
+            )
+          }
+          if (e.distributePayoutsEvent) {
+            elem = (
+              <DistributePayoutsElem
+                event={e.distributePayoutsEvent as DistributePayoutsEvent}
+              />
+            )
+          }
+          if (e.distributeReservedTokensEvent) {
+            elem = (
+              <DistributeReservedTokensEventElem
+                event={
+                  e.distributeReservedTokensEvent as DistributeReservedTokensEvent
+                }
+              />
+            )
+          }
+          if (e.useAllowanceEvent) {
+            // TODO
+            // elem = (
+            //   <DeployedERC20EventElem
+            //     event={e.deployedERC20Event as DeployedERC20Event}
+            //   />
+            // )
+          }
+
+          if (!elem) return null
+
+          return (
+            <div
+              key={e.id}
+              style={{
+                marginBottom: 20,
+                paddingBottom: 20,
+                borderBottom: '1px solid ' + colors.stroke.tertiary,
+              }}
+            >
+              {elem}
+            </div>
+          )
+        }),
+      ),
+    [colors, projectEvents],
+  )
+
+  const listStatus = useMemo(() => {
+    const count =
+      projectEvents?.pages?.reduce((prev, cur) => prev + cur.length, 0) ?? 0
+
+    if (isLoading || isFetchingNextPage) {
+      return (
+        <div>
+          <Loading />
+        </div>
+      )
+    } else if (count === 0 && !isLoading) {
```

Comment:
**nitpick (non-blocking)** these can become their own `if` because we're returning in each one.